draft-ietf-anima-brski-prm.md

```diff
@@ -336,84 +320,84 @@ Solution examples based on existing technology are provided with the focus on ex
+    +---- Drop Ship -----| Vendor Service            |
+    |                    +---------------+-----------+
+    |                    | M anufacturer |           |
+    |                    | A uthorized   | Ownership |
+    |                    | S igning      | Tracker   |
+    |                    | A uthority    |           |
+    |                    +---------------+-----------+
+    |                                         ^
+    |                                         | BRSKI-
+    |                                         | MASA
+    |          ...............................|.........
+    V          .                              v        .
++--------+     .  +------------+        +-----------+  .
+|        |     .  |            |        |           |  .
+| Pledge | BRSKI- | Registrar- | BRSKI- | Domain    |  .
+|        |  PRM   | Agent      |  EST   | Registrar |  .
+|        |<------>|            |<------>| (PKI RA)  |  .
+|        |     .  |     LDevID |        |           |  .
+|        |     .  +------------+        +-----+-----+  .
+| IDevID |     .                              |        .
+|        |     .           +------------------+-----+  .
++--------+     .           | Key Infrastructure     |  .
+               .           | (e.g., PKI Certificate |  .
+               .           |        Authority)      |  .
+               .           +------------------------+  .
+               .........................................
+                         "Domain" Components
 ~~~~
-{: #uc2figure title='BRSKI-PRM Architecture overview using registrar-agent' artwork-align="left"}
+{: #uc2figure title='BRSKI-PRM architecture overview using registrar-agent' artwork-align="left"}
 
 The following list describes the components in a (customer) site domain:
 
 * Pledge: The pledge is expected to respond with the necessary data objects for bootstrapping to the registrar-agent.
   The protocol used between the pledge and the registrar-agent is assumed to be HTTP in the context of this document.
-  Other protocols may be used like CoAP, Bluetooth, or NFC, but are out of scope of this document.
-  A pledge acting as a server during bootstrapping leads to some differences to BRSKI:
+  Other protocols such as CoAP, NFC, or Bluetooth may be used, but are out of scope of this document.
+  A pledge acting as a server during bootstrapping leads to some differences for BRSKI:
 
   * Discovery of the pledge by the registrar-agent must be possible.
 
   * As the registrar-agent must be able to request data objects for bootstrapping of the pledge, the pledge must offer corresponding endpoints.
 
-  * The registrar-agent MAY provide additional data to the pledge in the context of the voucher triggering request, that the pledge includes into the PVR.
+  * The registrar-agent MAY provide additional data to the pledge in the context of the voucher-request trigger, which the pledge includes into the PVR.
```

Comment:
MAY include, MUST include? As it appears to be for identification, it should be a MUST.

Reply:
Yes, it is a MUST, we used to have also the registrar-agent certificate here as option but removed it and did not change the text here.

Reply:
I was thinking about the statement for the pledge: I think it is most important that the pledge includes whatever information the reg-agent provides in the PVR. Hence, "which the pledge MUST include into the PVR."

For the reg-agent I am not sure if it MUST include information for PRM to work or if it MAY include any kind of information. If it MUST include it for PRM to work, then you MUST ;) state what information this is. Maybe this is again actually something for Section 6. (I see Section 5 for the normative statements about component internals (like behavior) and their interfaces, Section 6 about what is exchanged how between components.

Reply:
Yes, section 6 defines, that the registrar-agent MUST include the agent-signed-data to enable the registrar to perform the verification that the right registrar-agent was in place.
Adopted the sentence to 
 _The registrar-agent MUST provide additional data to the pledge in the context of the voucher-request trigger, which the pledge includes into the PVR as defined in section 6.1.1._